File: app/src/main/java/se/curtrune/lucy/workers/InternetWorker.java
package se.curtrune.lucy.workers;

import static androidx.core.content.ContextCompat.getSystemService;
import static se.curtrune.lucy.util.Logger.log;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.Network;
import android.net.NetworkInfo;

public class InternetWorker {
    public static boolean VERBOSE = false;
    public static boolean isConnected(Context context){
        if( VERBOSE) log("InternetWorker.isConnected()");
        ConnectivityManager connectivityManager  = getSystemService(context, ConnectivityManager.class);
        //log( connectivityManager != null ? "...got connectivityManager !": "no connectivity manager");
        if (connectivityManager == null){
            log("...unable to get connectivityManager, assuming not connected");
            return false;
        }
        NetworkInfo networkInfo =  connectivityManager.getActiveNetworkInfo();
        if( networkInfo == null){
            log("...networkInfo == null, assuming not connected");
            return false;
        }
        return connectivityManager.getActiveNetworkInfo().isConnectedOrConnecting();
    }
}

File: app/src/main/java/se/curtrune/lucy/workers/InternetWorker.kt
package se.curtrune.lucy.workers

import android.app.Application
import android.net.ConnectivityManager
import androidx.core.content.ContextCompat
import se.curtrune.lucy.util.Logger

class InternetWorker(val application: Application){
    var VERBOSE: Boolean = false
    fun isConnected(): Boolean {
        if (VERBOSE) Logger.log("InternetWorker.isConnected()")
        val connectivityManager = ContextCompat.getSystemService(
            application.applicationContext,
            ConnectivityManager::class.java
        )
        if (connectivityManager == null) {
            println("...unable to get connectivityManager, assuming not connected")
            return false
        }
        val networkInfo = connectivityManager.activeNetworkInfo
        if (networkInfo == null) {
            println("...networkInfo == null, assuming not connected")
            return false
        }
        return connectivityManager.activeNetworkInfo!!.isConnectedOrConnecting
    }
}
